frontend/public/components/factory/ListPage/filter-hook.ts

```diff
@@ -0,0 +1,39 @@
+import * as React from 'react';
+
+import { getAllTableFilters } from '../table';
+import { RowFilter } from '../../filter-toolbar';
+
+export type OnFilterChange = (type: string, value: string | string[]) => void;
+
+const filterData = <D>(
+  data: D[],
+  filter: { [key: string]: string | string[] } = {},
+  rowFilters?: RowFilter[],
```

Comment:
```suggestion
  rowFilters?: RowFilter[],
```
Is this really `RowFilter` ? 
Because `rowFilters[type]` should give undefined as `type` is `string`?

Reply:
good catch!. The type was wrong, fixed